xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -128,6 +128,7 @@ void DX::DeviceResources::GetDisplayMode(DXGI_MODE_DESC* mode) const
   ComPtr<IDXGIOutput> pOutput;
 
   m_swapChain->GetContainingOutput(&pOutput);
+  if (pOutput != nullptr) //*FIXME! During MultiGPU screen switching fall to last working renderer*//
```

Comment:
I didn't get this. the code below has undefined behavior in case then `pOutput` equals nullptr

Reply:
First, thank you for patience.. Eh Eh.. 
On monogpu systems that never go to nullptr, and if it happens you'll have a beautiful access violation crash cause trying to reading on null. That's is what happens in multigpu if you set a non primary adapter. On multi gpu, if after starting app you don't change screen in setting, happens the same of mono gpu. If you switch adapter and start rendering in theory we have to reinitialize with created3d11 and consequent calls, if we puts a condition here, first we avoid the exception and the consequent crash, second we leave swapchain pointed to the first adapter, the thing continue to works with new screen and old renderer. 
For avoiding this hack we need to reinitialize d3d11createdevice after screen changing.
It's absolutely safe for monogpu system

Reply:
If we will recreate a device on changing adapter then `GetContainingOutput` will always return valid pointer.